winml/lib/Api.Core/WinMLAdapter.cpp

```diff
@@ -727,10 +671,14 @@ void STDMETHODCALLTYPE InferenceSession::ReleaseCompletedReferences(onnxruntime:
 }
 
 HRESULT STDMETHODCALLTYPE InferenceSession::CopyOneInputAcrossDevices(
-  const char* input_name,
-  const OrtValue* orig_mlvalue, 
-  OrtValue** new_mlvalue) {
-  return E_NOTIMPL;
+    const char* input_name,
+    const OrtValue* orig_mlvalue,
+    OrtValue** new_mlvalue) {
+  auto session_protected_load_accessor =
+      static_cast<InferenceSessionProtectedLoadAccessor*>(session_.get());
+  const onnxruntime::SessionState& sessionState = session_protected_load_accessor->GetSessionState();
+  *new_mlvalue = new OrtValue;
```

Comment:
>ew OrtValue; [](start = 18, length = 12)

this will leak if CopyOneInputAcrossDevices fails.    instead you can use a smart ptr and then release it into the output buffer when we are all ready.

Reply:
Good point!